layers/gpu/error_message/gpuav_error_message.cpp

```diff
@@ -588,45 +417,215 @@ bool Validator::AnalyzeAndGenerateMessage(VkCommandBuffer cmd_buffer, const LogO
+               << HandleToUint64(tracker_info->pipeline) << ")\n";
+            if (tracker_info->shader_module == gpu::kPipelineStageInfoHandle) {
+                ss << "Shader Module was passed in via VkPipelineShaderStageCreateInfo::pNext\n";
+            } else {
+                ss << "Shader Module " << LookupDebugUtilsNameNoLock(debug_report, HandleToUint64(tracker_info->shader_module))
+                   << "(" << HandleToUint64(tracker_info->shader_module) << ")\n";
+            }
+        }
+    }
+
+    ss << std::dec << std::noshowbase;
+    ss << "SPIR-V Instruction Index = " << instruction_position << "\n";
+
+    // Find the OpLine just before the failing instruction indicated by the debug info.
+    // SPIR-V can only be iterated in the forward direction due to its opcode/length encoding.
+    uint32_t index = 0;
+    uint32_t reported_file_id = 0;
+    uint32_t reported_line_number = 0;
+    uint32_t reported_column_number = 0;
+    for (const spirv::Instruction &insn : instructions) {
+        if (insn.Opcode() == spv::OpLine) {
+            reported_file_id = insn.Word(1);
+            reported_line_number = insn.Word(2);
+            reported_column_number = insn.Word(3);
+        }
+        if (index == instruction_position) {
+            break;
+        }
+        index++;
+    }
+
+    if (reported_file_id == 0) {
+        ss << "Unable to find SPIR-V OpLine for source information.  Build shader with debug info to get source information.\n";
+        return ss.str();
+    }
+
+    std::string prefix;
+    if (container_type == LayerObjectTypeDebugPrintf) {
+        prefix = "Debug shader printf message generated ";
+    } else {
+        prefix = "Shader validation error occurred ";
+    }
+
+    // Create message with file information obtained from the OpString pointed to by the discovered OpLine.
+    bool found_opstring = false;
+    std::string reported_filename;
+    for (const spirv::Instruction &insn : instructions) {
+        if (insn.Opcode() == spv::OpString && insn.Length() >= 3 && insn.Word(1) == reported_file_id) {
+            found_opstring = true;
+            reported_filename = insn.GetAsString(2);
+            if (reported_filename.empty()) {
+                ss << prefix << "at line " << reported_line_number;
+            } else {
+                ss << prefix << "in file " << reported_filename << " at line " << reported_line_number;
+            }
+            if (reported_column_number > 0) {
+                ss << ", column " << reported_column_number;
+            }
+            ss << "\n";
+            break;
+        }
+        // OpString can only be in the debug section, so can break early if not found
+        if (insn.Opcode() == spv::OpFunction) break;
+    }
+
+    if (!found_opstring) {
+        ss << "Unable to find SPIR-V OpString for file id " << reported_file_id << " from OpLine instruction.\n";
```

Comment:
```suggestion
        ss << "Unable to find SPIR-V OpString for file from OpLine instruction.\n";
```
file id already reported below